json-schema/Request-Context-v2.json

```diff
@@ -0,0 +1,267 @@
+  "type": "object",
+  "properties": {
+    "userIdentifier": {
+      "type": "string",
+      "description": "The alphanumeric user id as known to the system they have signed into (which is consuming the FHIR API)",
+      "example": "LJRHODE"
+    },
+    "userRole": {
+      "type": "string",
+      "enum": [
+          "PAT",
+          "PROV"
+      ]
+    },
+    "hpiPractitioner": {
+      "type": "object",
+      "description": "User health practitioner identification as an HPI FHIR reference",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Fixed to refer to FHIR Practitioner resource type",
+          "example": "Practitioner",
+          "pattern": "^Practitioner$"
+        },
+        "identifier": {
+          "type": "object",
+          "description": "These attributes constitute a FHIR Identifier for an HPI Practitioner resource",
+          "properties": {
+            "use": {
+              "type": "string",
+              "description": "Fixed to official",
+              "const": "official"
+            },
+            "system": {
+              "type": "string",
+              "description": "Url that identifies a HPI Practitioner resource",
+              "const": "https://standards.digital.health.nz/ns/hpi-person-id"
+            },
+            "value": {
+              "type": "string",
+              "description": "The practitioner's Common Person Number or CPN as defined by the HPI Practitioner Index) in the form NNXXXX where N is numeric and X is Alphabetic",
+              "pattern": "^[0-9]{2}[A-Z0-9]{4}$",
+              "example": "91ZABY"
+            }
+          },
+          "required": [ "use", "system", "value"]
+        },
+        "display": {
+          "type": "string",
+          "description": "The full name of the practitioner", 
+          "example": "Lyndi Jane Rhode"
+        }
+      },
+      "required": ["type","identifier","display"]
+    },          
+    "registrationAuthorityNumber": {
+      "type": "object",
+      "description": "These three attributes constitute a FHIR Identifier defining the registration number issued to a practitioner by their RA",
+      "properties": {
+        "use": {
+          "type": "string",
+          "description": "Fixed to official",
+          "const": "official"
+        },
+        "system": {
+          "type": "string",
+          "description": "Url identifying the practitioner's registration authority/council from https://standards.digital.health.nz/ns/hpi-ra-identifier-code",
+          "example": "https://standards.digital.health.nz/ns/medical-council-id"
+        },
+        "value": {
+          "type": "string",
+          "description": "Number issued to the practitioner by their registration authority",
+          "example": "123456"
+        }
+      },
+      "required": ["use","system","value"]
+    },
+    "hpiFacility": {
+      "type": "object",
```

Comment:
This isn't required to be an object, imo. the field is defined as `hpiFacility` which explicitly already contains, the `type`, `system` and implictly the `use`. This can be `"type":"string"` with the value of the HPI-F.

Reply:
I think `hpiFacility` and `hpiOrganisation` are better names than the current `orgIdentifier` and `facilityIdentifier`, I agree this should replace them.